src/compiler/scala/tools/nsc/transform/patmat/PatternMatching.scala

```diff
@@ -222,7 +222,7 @@ trait Interface extends ast.TreeDSL {
         object substIdentsForTrees extends Transformer {
           private def typedIfOrigTyped(to: Tree, origTp: Type): Tree =
             if (origTp == null || origTp == NoType) to
-            // important: only type when actually substing and when original tree was typed
+            // important: only type when actually substring and when original tree was typed
```

Comment:
Actually, that's not a typo, "substing" is "substituting" here, not "substring".